src/RequestInformation.cs

```diff
@@ -101,10 +104,14 @@ public Uri URI
         /// Vanity method to add the query parameters to the request query parameters dictionary.
         /// </summary>
         /// <param name="source">The query parameters to add.</param>
-        public void AddQueryParameters(object source)
+#if NET5_0_OR_GREATER
+        public void AddQueryParameters<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicProperties)] T>(T source)
```

Comment:
this could be breaking to people who:
- target net 5 or above
- call the method directly
- have enabled trimming

But that population is small and would not have any query parameter work properly, so I think we could rule this as a fix.

Reply:
Just to confirm, will this break generated code that does not specify the type parameter? For example https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/5fbfa66e9b0d7f97228ab7a99dec749f90f71810/src/Microsoft.Graph/Generated/Users/UsersRequestBuilder.cs#L126C62-L126C77

Reply:
no, the type parameter should be inferred from the first argument. I don't expect code generation to need any change.

Reply:
That makes sense. Thanks for confirming.